src/Sylius/Bundle/CoreBundle/Tests/Controller/CurrencyApiTest.php

```diff
@@ -0,0 +1,247 @@
+
+        $response = $this->client->getResponse();
+
+        $this->assertResponse($response, 'currency/create_response', Response::HTTP_CREATED);
+    }
+
+    public function testGetCurrenciesListAccessDeniedResponse()
+    {
+        $this->client->request('GET', '/api/currency/');
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'authentication/access_denied_response', Response::HTTP_UNAUTHORIZED);
+    }
+
+    public function testGetCurrenciesListResponse()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $this->loadFixturesFromFile('resources/currencies.yml');
+
+        $this->client->request('GET', '/api/currency/', [], [], static::$authorizedHeader);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'currency/index_response', Response::HTTP_OK);
+    }
+
+    public function testGetCurrencyAccessDeniedResponse()
+    {
+        $this->client->request('GET', '/api/currency/1');
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'authentication/access_denied_response', Response::HTTP_UNAUTHORIZED);
+    }
+
+    public function testGetCurrencyWhichDoesNotExistResponse()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+
+        $this->client->request('GET', '/api/currency/-1', [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'error/not_found_response', Response::HTTP_NOT_FOUND);
+    }
+
+    public function testGetCurrencyResponse()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $currencies = $this->loadFixturesFromFile('resources/currencies.yml');
+
+        $this->client->request('GET', '/api/currency/'.$currencies['currency_1']->getId(), [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'currency/show_response', Response::HTTP_OK);
+    }
+
+    public function testFullUpdateCurrencyAccessDeniedResponse()
+    {
+        $this->client->request('PUT', '/api/currency/1');
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'authentication/access_denied_response', Response::HTTP_UNAUTHORIZED);
+    }
+
+    public function testFullUpdateCurrencyWhichDoesNotExistResponse()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+
+        $this->client->request('PUT', '/api/currency/-1', [], [], static::$authorizedHeaderWithContentType);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'error/not_found_response', Response::HTTP_NOT_FOUND);
+    }
+
+    public function testFullUpdateCurrencyValidationFailResponse()
```

Comment:
This test still fails. The code field doesn't raise a constraint violation for some reason. Anyone an idea as to why?